runtimeconfig/manager_test.go

```diff
@@ -419,3 +436,37 @@ func TestManager_ShouldFastFailOnInvalidConfigAtStartup(t *testing.T) {
 	require.NoError(t, err)
 	require.Error(t, services.StartAndAwaitRunning(context.Background(), m))
 }
+
+func TestManager_UnchangedFileDoesntTriggerReload(t *testing.T) {
+	loadCounter := atomic.NewInt32(0)
+
+	cfg := newTestOverridesManagerConfig(t, 100*time.Millisecond, func(reader io.Reader) (interface{}, error) {
+		loadCounter.Inc()
+		return valueLoader(reader)
+	})
+
+	overridesManager, err := New(cfg, nil, log.NewNopLogger())
+	require.NoError(t, err)
+
+	ch := overridesManager.CreateListenerChannel(10) // must be big enough to hold all modifications.
+
+	require.NoError(t, services.StartAndAwaitRunning(context.Background(), overridesManager))
+
+	test.Poll(t, time.Second, 1, func() interface{} {
+		return int(loadCounter.Load())
+	})
+
+	// Let's make some modifications to the config
+	const mods = 3
+	const modDelay = 500 * time.Millisecond
+	for i := 0; i < mods; i++ {
+		writeValueToFile(t, cfg.LoadPath[0], value{Value: i})
+		// wait before next rewrite, but also after last rewrite to give manager a chance to reload the file again
+		time.Sleep(modDelay)
+	}
+
+	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), overridesManager))
+
+	assert.Equal(t, int(loadCounter.Load()), mods+1)  // + 1 for initial load, before modifications
+	assert.Equal(t, int(loadCounter.Load()), len(ch)) // Loaded values
```

Comment:
The expectation should come before the tested value:
```suggestion
	assert.Equal(t, mods+1, int(loadCounter.Load()))  // + 1 for initial load, before modifications
	assert.Equal(t, len(ch), int(loadCounter.Load())) // Loaded values
```

Reply:
👍  Second line actually tests that `len(ch)` is correct.

Reply:
@pstibrany aha, was difficult to tell what's the expectation on that line :D

Reply:
I would then suggest to make it, since the number of loads could be wrong (in case of a bug):

```
assert.Equal(t, mods+1, len(ch)) // Loaded values
```

Reply:
Updated my suggestion to reflect the above :point_up: